src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -0,0 +1,287 @@
+package org.broadinstitute.hellbender.utils.pairhmm;
+
+import org.apache.log4j.Logger;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.genotyper.ReadLikelihoods;
+import org.broadinstitute.hellbender.utils.genotyper.LikelihoodMatrix;
+import org.broadinstitute.hellbender.utils.haplotype.Haplotype;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class VectorLoglessPairHMM extends LoglessPairHMM {
+
+    protected final static Logger logger = Logger.getLogger(VectorLoglessPairHMM.class);
+    protected long threadLocalSetupTimeDiff = 0;
+    protected static long pairHMMSetupTime = 0;
+    protected final static Boolean runningOnMac = System.getProperty("os.name").toLowerCase().startsWith("mac");
+
+    //Used to copy references to byteArrays to JNI from reads
+    protected class JNIReadDataHolderClass {
+        public byte[] readBases = null;
+        public byte[] readQuals = null;
+        public byte[] insertionGOP = null;
+        public byte[] deletionGOP = null;
+        public byte[] overallGCP = null;
+    }
+
+    //Used to copy references to byteArrays to JNI from haplotypes
+    protected class JNIHaplotypeDataHolderClass {
+        public byte[] haplotypeBases = null;
+    }
+
+    /**
+     * Function to initialize the fields of JNIReadDataHolderClass and JNIHaplotypeDataHolderClass from JVM.
+     * C++ codegets FieldIDs for these classes once and re-uses these IDs for the remainder of the program. Field IDs do not
+     * change per JVM session
+     *
+     * @param readDataHolderClass      class type of JNIReadDataHolderClass
+     * @param haplotypeDataHolderClass class type of JNIHaplotypeDataHolderClass
+     */
+    private native void jniInitializeClassFields(Class<?> readDataHolderClass, Class<?> haplotypeDataHolderClass);
+
+
+    public Boolean isAVXSupported() {
+        // use a grep command to check for AVX support
+        // grep exit code = 0 if a match was found
+        String command = runningOnMac ? "sysctl -a | grep machdep.cpu.features | grep -i avx" :
+                                        "grep -i avx /proc/cpuinfo";
+        try {
+            Process process = new ProcessBuilder("/bin/sh", "-c", command).start();
+            if (process.waitFor() != 0) {
+                return false;
+            }
+            if (process.exitValue() != 0) {
+                return false;
+            }
+        }
+        catch (InterruptedException | IOException e) {
```

Comment:
Could you output a log message in these cases?  It would help resolve the confusing case of someone having avx support but no grep or something bizarre like that